### PR TITLE
Fix insert after create in memory-connector

### PR DESCRIPTION
**Pull request by ebyhr**
#7733

### CODE REVIEW

File: presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryPageSinkProvider.java [hunk 1]
```diff
@@ -65,6 +65,7 @@ public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHa
         checkState(memoryInsertTableHandle.getActiveTableIds().contains(tableId));
 
         pagesStore.cleanUp(memoryInsertTableHandle.getActiveTableIds());
+        pagesStore.initialize(tableId);
```
Comment (kokosing) [inline presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryPageSinkProvider.java]: This seems to be a bug. Please consider following scenario, there had been some write query and so worker stored data for this table, then there was a crash and worker got restarted. Now, with your change, when a new insert comes, table will be initialized again and nobody would notice the data loss.

I wish we had tests for that.

File: presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryPageSourceProvider.java [hunk 2]
```diff
@@ -55,6 +55,7 @@ public ConnectorPageSource createPageSource(
         List<Integer> columnIndexes = columns.stream()
                 .map(MemoryColumnHandle.class::cast)
                 .map(MemoryColumnHandle::getColumnIndex).collect(toList());
+        pagesStore.initialize(tableId);
```
Comment (kokosing) [inline presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryPageSourceProvider.java]: this seems to be a bug as well. Please consider similar scenario scenario as in previous comment. There had been some write query and so worker stored data for this table, then there was a crash and worker got restarted. Now, with your change, when select comes worker would just return no data in the same way as it there were never any data.

And once again, I wish we had tests for that.

**Comment by pnowojski** [comment]
There is another scenario for which I'm now worried, what if we create:
```
CREATE TABLE memory.default.nation AS SELECT * FROM hive.default.nation WHERE nationkey < 0;
```
and hive would return no splits for the source table. Do we have guarantee that Presto will schedule `TableWriterOperators` for memory connector on all nodes? Do we want to relay on such behavior? I think it will be cleaner to just implement `MemoryPageSink::finish()` as @kokosing wrote above.

One more scenario that I think could be broken at the moment, is when we have 1000 nodes cluster and `TableWriterOperators` for memory connector will be scheduled only on 100 of the nodes. Currently `MemoryMetadata::beginCreateTable` happily assumes that table will be initialized on all of the 1000 nodes. This also would be fixed with better `MemoryPageSink::finish()`.

CC @haozhun @electrum what do you think?

File: presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryPageSinkProvider.java (hunk 1 above)
Comment (ebyhr) [inline presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryPageSinkProvider.java]: Oh, I'm sorry... I might have tested with other branch and misunderstood all test succeeded.
Exactly this commit introduce a bug.

**Pull request closed without merging**